File: src/dataset/c/beer_song_Exercism_ThomasZumsteg/gemini/GeminiFlash_beer_song_v2.c
#include "beer_song.h"
#include <stdio.h>
#include <string.h>

#define BUFFER_SIZE   (1024)

void verse(char *buff, int v) {
    if (v == 0) {
        strcpy(buff, "No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n");
    } else if (v == 1) {
        strcpy(buff, "1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n");
    } else if (v == 2) {
        strcpy(buff, "2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n");
    } else {
        snprintf(buff, BUFFER_SIZE, "%d bottles of beer on the wall, %d bottles of beer.\nTake one down and pass it around, %d bottles of beer on the wall.\n", v, v, v-1);
    }
}

void sing(char *buff, int start, int stop) {
    char *current_pos = buff;
    for (int v = start; v >= stop; v--) {
        char verse_buff[BUFFER_SIZE];
        verse(verse_buff, v);
        size_t verse_len = strlen(verse_buff);
        memcpy(current_pos, verse_buff, verse_len);
        current_pos += verse_len;
        *current_pos = '\n';
        current_pos++;
    }
    if (current_pos > buff) {
        current_pos--;
    }
    *current_pos = '\0';
}

File: src/dataset/Java/beer_song_Exercism_java_rabestro/gemini/GeminiFlash_BeerSong_v2.java
class BeerSong {

    private static final int INITIAL_BOTTLES = 99;

    public String sing(int count, int repeat) {
        StringBuilder song = new StringBuilder(repeat * 250); // Pre-allocate StringBuilder size

        for (int i = 0; i < repeat; i++) {
            if (count == 0) {
                song.append("No more bottles of beer on the wall, no more bottles of beer.\n")
                        .append("Go to the store and buy some more, ")
                        .append(INITIAL_BOTTLES).append(" bottles of beer on the wall.\n\n");
                count = INITIAL_BOTTLES;
            } else {
                song.append(bottleText(count))
                        .append(" of beer on the wall, ")
                        .append(bottleText(count))
                        .append(" of beer.\nTake ")
                        .append(count == 1 ? "it" : "one")
                        .append(" down and pass it around, ");
                count--;
                song.append(bottleText(count)).append(" of beer on the wall.\n\n");
            }
        }
        return song.toString();
    }

    public String singSong() {
        return sing(INITIAL_BOTTLES, 100);
    }

    private String bottleText(int count) {
        if (count == 0) {
            return "no more bottles";
        } else if (count == 1) {
            return "1 bottle";
        } else {
            return Integer.toString(count) + " bottles";
        }
    }
}